### ISSUE TITLE
JSON serialization broken for BigDecimal

### ISSUE THREAD

**Issue by mkljakubowski** [playframework/playframework#6530]
```
class JsonBigDecimalTest extends Specification {

  case class Foo(bar: BigDecimal)

  object Foo {
    implicit val format = Json.format[Foo]
  }

  "json serialization" should {
    "serialize big bigdecimal" in {
      val foo = Foo(BigDecimal(2e128))
      val foostr = Json.toJson(foo).toString()
      Json.parse(foostr).as[Foo] should_== foo
    }

  }

}
```
This test fails with exception in play 2.4.6 but not in play 2.4.4.
```
[error]    com.fasterxml.jackson.databind.JsonMappingException: Illegal embedded sign character (DefaultSerializerProvider.java:141)
[error] com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:141)
[error] com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2444)
[error] play.api.libs.json.jackson.JacksonJson$.generateFromJsValue(JacksonJson.scala:249)
[error] play.api.libs.json.Json$.stringify(Json.scala:61)
[error] play.api.libs.json.JsValue$class.toString(JsValue.scala:16)
[error] play.api.libs.json.JsObject.toString(JsValue.scala:76)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:24)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:22)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:66)
[error] play.api.libs.json.jackson.JsValueSerializer$$anonfun$serialize$2.apply(JacksonJson.scala:81)
[error] play.api.libs.json.jackson.JsValueSerializer$$anonfun$serialize$2.apply(JacksonJson.scala:79)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:79)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:43)
[error] com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:130)
[error] com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2444)
[error] play.api.libs.json.jackson.JacksonJson$.generateFromJsValue(JacksonJson.scala:249)
[error] play.api.libs.json.Json$.stringify(Json.scala:61)
[error] play.api.libs.json.JsValue$class.toString(JsValue.scala:16)
[error] play.api.libs.json.JsObject.toString(JsValue.scala:76)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:24)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:22)
[error] CAUSED BY
[error]  java.lang.NumberFormatException: Illegal embedded sign character (JacksonJson.scala:66)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:66)
[error] play.api.libs.json.jackson.JsValueSerializer$$anonfun$serialize$2.apply(JacksonJson.scala:81)
[error] play.api.libs.json.jackson.JsValueSerializer$$anonfun$serialize$2.apply(JacksonJson.scala:79)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:79)
[error] play.api.libs.json.jackson.JsValueSerializer$.serialize(JacksonJson.scala:43)
[error] com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:130)
[error] com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2444)
[error] play.api.libs.json.jackson.JacksonJson$.generateFromJsValue(JacksonJson.scala:249)
[error] play.api.libs.json.Json$.stringify(Json.scala:61)
[error] play.api.libs.json.JsValue$class.toString(JsValue.scala:16)
[error] play.api.libs.json.JsObject.toString(JsValue.scala:76)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:24)
[error] xxx.JsonBigDecimalTest$$anonfun$3$$anonfun$apply$4.apply(JsonBigDecimalTest.scala:22)
```

**Comment by mkljakubowski** [comment]
this was broken by this commit https://github.com/playframework/playframework/commit/0f4d17e00d200e5718d5918fc7b5a3c3e7d29487

**Comment by gmethvin** [comment]
Fixed by #6531. Thanks @mkljakubowski!
